src/QueryEditor.test.tsx

```diff
@@ -95,10 +98,10 @@ describe('QueryEditor', () => {
       ...q,
       connectionArgs: { ...q.connectionArgs, database: 'foo' },
     });
-    expect(onRunQuery).toHaveBeenCalled();
+    expect(onRunQuery).not.toHaveBeenCalled();
```

Comment:
~remind me why onRunQuery shouldn't run? Do we have like an onRunAsyncQuery or something?~ nevermind I see below we only want to run it now when the run query button is clicked, totally makes sense! I _think_ this could be a separate PR if we wanted, again kind of nitpicking tho.

Reply:
will probably just leave this in this pr for now.